pyccel/codegen/printing/fcode.py

```diff
@@ -1193,7 +1195,18 @@ def _print_BooleanFalse(self, expr):
         return '.False.'
 
     def _print_String(self, expr):
-        formatted_str = expr.arg.replace("'","''")
+        # characters speciaux are not handled only newline
+        formatted_str = expr.arg.replace("'","''")\
+                                .replace('\a', '\\a')\
+                                .replace('\b', '\\b')\
+                                .replace('\f', '\\f')\
+                                .replace('\r', '\\r')\
+                                .replace('\t', '\\t')\
+                                .replace('\v', '\\v')
```

Comment:
By using the [`ACHAR`](https://gcc.gnu.org/onlinedocs/gfortran/ACHAR.html) Fortran function and this [ASCII table](https://en.wikipedia.org/wiki/Escape_sequences_in_C#Table_of_escape_sequences) it should be possible to handle all the special characters listed here. One first splits the string into substrings (either normal strings, or special characters), then replaces the special characters with their ASCII equivalent, and finally concatenates all substrings as done below for `new_line`.

Please try!

Reply:
done